subsys/bluetooth/mesh/net.c

```diff
@@ -108,36 +110,45 @@ K_MEM_SLAB_DEFINE(loopback_buf_pool,
 		  sizeof(struct loopback_buf),
 		  CONFIG_BT_MESH_LOOPBACK_BUFS, __alignof__(struct loopback_buf));
 
-static uint32_t dup_cache[CONFIG_BT_MESH_MSG_CACHE_SIZE];
-static int   dup_cache_next;
+static uint32_t dup_cache[CONFIG_BT_MESH_MSG_CACHE_SIZE]
+			 [CONFIG_BT_MESH_MSG_CACHE_SIZE];
+static struct {
+	uint8_t idx;
+	uint8_t next;
+} dup;
 
 static bool check_dup(struct net_buf_simple *data)
 {
 	const uint8_t *tail = net_buf_simple_tail(data);
 	uint32_t val;
+	uint8_t idx;
 	int i;
 
 	val = sys_get_be32(tail - 4) ^ sys_get_be32(tail - 8);
 
-	for (i = 0; i < ARRAY_SIZE(dup_cache); i++) {
-		if (dup_cache[i] == val) {
+	idx = val % ARRAY_SIZE(dup_cache[0]);
+
+	for (i = 0; i < ARRAY_SIZE(dup_cache[0]); i++) {
+		if (dup_cache[idx][i] == val) {
 			return true;
 		}
 	}
 
-	dup_cache[dup_cache_next++] = val;
-	dup_cache_next %= ARRAY_SIZE(dup_cache);
+	dup.idx	= idx;
+	dup.next %= ARRAY_SIZE(dup_cache[0]);
+	dup_cache[idx][dup.next++] = val;
```

Comment:
```suggestion
	dup_cache[idx][dup.next] = val;
	dup.next++;
```
Little bit easier to read, and more similar to line 164